### PR TITLE
[133] Provide Basic Documentation

### PR DESCRIPTION
**Pull request by elbenfreund**
This PR expands what little documentation we already had and adds several new pages to help potential contributors to get a quick overview about how things work as well as to codify best practices via the *Styleguide*.

I opted to use a single commit for each documentation page to be more verbose and transparent. If you @border0464111 feel that this is too much, please let me know.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -12,50 +11,49 @@ Types of Contributions
...
-* Remember that this is a volunteer-driven project, and that contributions
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Remember that this is a volunteer-driven project, and that contributions
   are welcome :)
 
 Get Started!
 ------------
+Ready to contribute? Here's how to set up `hamster-shell-extension` for local
+development.
```
Comment (border0464111) [inline CONTRIBUTING.rst]: WISHLIST:  It need a link here to some tutorial/doc on about   "how to set up `hamster-shell-extension` for local
 development."

**Comment by border0464111** [comment]
As it's a single PR it dosen't allow much more granularity to do it as one commit or not. So It's up to you :)

**Comment by elbenfreund** [comment]
Not at all! Please feel free to use githubs ability to comment individual lines of a PR to add your feedback!
So if you have criticism about a particual commit, feedback is always welcome! If push comes to shove we I am on the liveral end of rewritiing git history (via a rebase).

File: CONTRIBUTING.rst (hunk 1 above)
Comment (elbenfreund) [inline CONTRIBUTING.rst]: Please feel encourage to create a new issue :)

**Pull request merged**